Source/DafnyCore/AST/Grammar/ProgramParser.cs

```diff
@@ -102,6 +104,23 @@ public class ProgramParser {
 
     ShowWarningsForIncludeCycles(program);
 
+    compilation.TranslationRecord = new TranslationRecord();
+    var records = options.Get(CommonOptionBag.TranslationRecords);
+    if (records != null) {
+      foreach (var path in options.Get(CommonOptionBag.TranslationRecords)) {
+        using var reader = new StreamReader(path.FullName);
+        var pathForErrors = options.GetPrintPath(path.FullName);
+        try {
+          var libraryRecord = TranslationRecord.Read(reader);
+          if (libraryRecord.Validate(errorReporter, pathForErrors, options, Token.NoToken)) {
```

Comment:
I think calling `options.GetPrintPath` could/should be moved into `.Validate`, since it also gets an `options` and that leaves less responsibility on the caller.